awscrt/http.py

```diff
@@ -0,0 +1,174 @@
+
+        for slot in self.__slots__:
+            setattr(self, slot, None)
+
+        self._bootstrap = bootstrap
+        self._tls_connection_options = tls_connection_options
+        self._on_connection_shutdown = on_connection_shutdown
+        self._native_handle = None
+
+    @staticmethod
+    def new_connection(bootstrap, host_name, port, socket_options, on_connection_shutdown, tls_connection_options):
+        """
+        Initiates a new connection to host_name and port using socket_options and tls_connection_options if supplied.
+        if tls_connection_options is None, then the connection will be attempted over plain-text.
+        on_connection_shutdown will be invoked if the connection shuts-down while you still hold a reference to it.
+        on_connection_shutdown takes a single argument of int type, to specify the shutdown reason.
+
+        returns a future where the result is a new instance to HttpClientConnection, once the connection has completed
+        and is ready for use.
+        """
+        assert tls_connection_options is None or isinstance(tls_connection_options, TlsConnectionOptions)
+        assert host_name is not None
+        assert port is not None
+        assert socket_options is not None and isinstance(socket_options, SocketOptions)
+        assert on_connection_shutdown is not None
+
+        future = Future()
+        connection = HttpClientConnection(bootstrap, on_connection_shutdown, tls_connection_options)
+
+        def on_connection_setup_native_cb(native_handle, error_code):
+            if error_code == 0:
+                connection._native_handle = native_handle
+                future.set_result(connection)
+            else:
+                future.set_exception(Exception("Error during connect: err={}".format(error_code)))
+
+        try:
+            if tls_connection_options:
+                internal_conn_options_handle = tls_connection_options._internal_tls_conn_options
+            else:
+                internal_conn_options_handle = None
+
+            _aws_crt_python.aws_py_http_client_connection_create(bootstrap._internal_bootstrap,
+                                                                 on_connection_setup_native_cb,
+                                                                 connection._on_connection_shutdown,
+                                                                 host_name,
+                                                                 port,
+                                                                 socket_options,
+                                                                 internal_conn_options_handle)
+
+        except Exception as e:
+            future.set_exception(e)
+
+        return future
+
+    def close(self):
+        """
+        Closes the connection, if you hold a reference to this instance of HttpClientConnection, on_connection_shutdown
+        will be invoked upon completion of the connection close.
```

Comment:
the "if you hold a reference to this instance of HttpClientConnection" part of this comment seems unnecessary. If you're calling a member function, you have a reference